packages/react/checkbox/src/Checkbox.tsx

```diff
@@ -84,28 +85,27 @@ const Checkbox = React.forwardRef((props, forwardedRef) => {
         ref={composedRefs}
         onClick={composeEventHandlers(props.onClick, (event) => {
           setChecked((prevChecked) => (prevChecked === 'indeterminate' ? true : !prevChecked));
-          hasConsumerStoppedPropagationRef.current = event.isPropagationStopped();
-          // Stop propagation from the button so that we only propagate one click event (from the input).
-          // We propagate changes from an input so that form events reflect checkbox updates.
-          if (!hasConsumerStoppedPropagationRef.current) event.stopPropagation();
+          if (isFormControl) {
+            hasConsumerStoppedPropagationRef.current = event.isPropagationStopped();
+            // if checkbox is in a form, stop propagation from the button so that we only propagate
+            // one click event (from the input). We propagate changes from an input so that native
+            // form validation works and form events reflect checkbox updates.
+            if (!hasConsumerStoppedPropagationRef.current) event.stopPropagation();
+          }
         })}
       />
-      <BubbleInput
-        stoppedPropagation={hasConsumerStoppedPropagationRef.current}
-        name={name}
-        value={value}
-        checked={checked}
-        required={required}
-        disabled={disabled}
-        style={{
-          position: 'absolute',
-          pointerEvents: 'none',
-          opacity: 0,
-          margin: 0,
-          transform: 'translateX(-100%)',
-          ...buttonSize,
-        }}
-      />
+      {isFormControl && (
+        <BubbleInput
+          control={button}
+          stoppedPropagation={hasConsumerStoppedPropagationRef.current}
+          name={name}
+          value={value}
+          checked={checked}
+          required={required}
+          disabled={disabled}
+          style={{ transform: 'translateX(-100%)' }}
```

Comment:
I kept this style up here because it is only needed when `BubbleInput` is rendered **after** the button. It seems this level should be in control of that style in case we move it before the button later.

Reply:
I agree with the place the code lives in but I think it would be nice to add a comment perhaps now that it's disconnected from the rest of the styles.